sonarqube/httpHelpers_test.go

```diff
@@ -0,0 +1,29 @@
+package sonarqube
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+// TestCensorError calls censorError with sample log messages , checking
+// for a valid return value.
+func TestCensorError(t *testing.T) {
+	cases := []struct {
+		errorMessage error
+		token        string
+		expected     string
+	}{
+		{
+			errorMessage: errors.New("Error updating SonarQube user: failed to execute http request: POST https://PASSWORD:@sonarqube.example.com/api/users/update_identity_provider?login=gitlab-john-doe&newExternalIdentity=john-doe&newExternalProvider=gitlab giving up after 5 attempt(s). Request: &{0xab1940 0xc00021c600}"),
+			token:        "PASSWORD",
+			expected:     "Error updating SonarQube user: failed to execute http request: POST https://********:@sonarqube.example.com/api/users/update_identity_provider?login=gitlab-john-doe&newExternalIdentity=john-doe&newExternalProvider=gitlab giving up after 5 attempt(s). Request: &{0xab1940 0xc00021c600}",
```

Comment:
I'm not sure I would have such a "complex" string in the test - things like the number of attempts, the request id etc are not relevant to the test - wouldn't a simple GET request to an endpoint suffice?
I'm a big fan of keeping tests simple.

I would also argue that the test is a bit brittle because it is testing too much.
Is the purpose of the test to ensure that the sensitive information is masked out or to ensure that the function returns the original error message but with the sensitive info masked?
I would argue the former and therefore all you should be testing here is that the sensitive info is NOT included in the output - this is the key part of the functionality and should be tested in isolation IMO.

You don't want the test to fail because for example some future refactor truncated the message.  That's a different problem and would have a different test :)

And indeed ideally you would have a test which has no sensitive info to check that you do get back the original string.